lib/api.js

```diff
@@ -46,6 +47,16 @@ class Madge {
 		}
 
 		this.config = Object.assign({}, defaultConfig, config);
+		if (typeof this.config.tsConfig === 'string') {
+			const ts = require('typescript');
+			const tsParsedConfig = ts.readJsonConfigFile(this.config.tsConfig, ts.sys.readFile);
+			const obj = ts.parseJsonSourceFileConfigFileContent(tsParsedConfig, ts.sys, path_.dirname(config.tsConfig));
```

Comment:
Nice moving this from `cli.js`!
One thing to confirm: Will `ts.readJsonConfigFile` throw error on cases of empty file name, file not exist, wrong file format, etc?

Reply:
Yes, I tried it in a local environment, and an error occurred within TypeScript.

Reply:
I didn't try it, maybe we can add a repo to test madge later.
The only thing I'm not sure was this, and since you tested I think we can proceed and including this in v6.0